src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js

```diff
@@ -1187,3 +1192,84 @@ describe('Add set mappings to concept', () => {
     });
   });
 });
+
+describe('buildNewMappingData', () => {
+  const fromConceptUrl = '/test/from/concept/url';
+  const toSourceUrl = '/test/to/source/url';
+  const toConceptCode = 'testCode';
+  const toConceptName = 'testName';
+  const mapType = MAP_TYPES_DEFAULTS[0];
+
+  it('should return the right mapping data if given an external concept', () => {
+    const mapping = {
+      source: INTERNAL_MAPPING_DEFAULT_SOURCE,
+      map_type: mapType,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    const expectedMapping = {
+      map_type: mapType,
+      from_concept_url: fromConceptUrl,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    expect(buildNewMappingData(mapping, fromConceptUrl)).toEqual(expectedMapping);
+  });
+
+  it('should return the right mapping data if given an internal concept', () => {
```

Comment:
How different is this from the last test?

Reply:
The source for the mapping is different and hence different data is returned as evidenced by the differing expectedMappings. Line 1205 and Line 1223.

Reply:
Is that why the description is the same?

Reply:
The description is not the same.
> should return the right mapping data if given an internal concept
> should return the right mapping data if given an external concept

Reply:
should return the right mapping data if given an internal concept
should return the right mapping data if given an internal concept

Reply:
Are we looking at the same file? 🙂
<img width="577" alt="image" src="https://user-images.githubusercontent.com/12200860/58267188-c759fb80-7d8b-11e9-8fa9-f9dfba1d2753.png">

Reply:
https://github.com/openmrs/openmrs-ocl-client/pull/455/files

Reply:
Lines 1231 and 1249 are not the same Daniel, are they?
Could you kindly point out the line numbers, perhaps therein lies the confusion.

Reply:
1249, 1289

Reply:
Thank you. These tests are each for different methods(and are in different describe blocks). One for `buildUpdateMappingData` and the other for `buildNewMappingData` which return distinct types of data.

Reply:
Why do they have the same description is they are testing different things?

Reply:
The tests on 1249 and 1289 test that those methods `should return the right mapping data if given an internal concept`. The meaning of `right mapping data` defers for each of those methods but that name still stands. Perhaps if you took a look at what those methods do this would be more clear?

They are testing the ~same thing on different methods.

Reply:
The fact that am asking this should trigger in your mind that latter maintainers of this code will struggle to understand why you used the same description for different things.

Reply:
Noted. Thanks for pointing this out. Updated these descriptions to explicitly state what each is testing.